lms/djangoapps/grades/course_grades.py

```diff
@@ -2,29 +2,17 @@
 Functionality for course-level grades.
 """
 from logging import getLogger
-from django.conf import settings
 import dogstats_wrapper as dog_stats_api
-import random
 
 from opaque_keys.edx.keys import CourseKey
-from openedx.core.djangoapps.signals.signals import GRADES_UPDATED
-
-from course_blocks.api import get_course_blocks
 from courseware.courses import get_course_by_id
-from courseware.model_data import ScoresClient
-from student.models import anonymous_id_for_user
-from util.db import outer_atomic
-from xmodule import graders, block_metadata_utils
-from xmodule.graders import Score
-
-from .context import grading_context
-from .scores import get_score
+from .new.course_grade import CourseGradeFactory
 
 
 log = getLogger(__name__)
 
 
-def iterate_grades_for(course_or_id, students, keep_raw_scores=False):
+def iterate_grades_for(course_or_id, students):
     """Given a course_id and an iterable of students (User), yield a tuple of:
 
     (student, gradeset, err_msg) for every student enrolled in the course.
```

Comment:
Should we return a namedtuple for this instead?  It can help clarify the interface & reduce potential for bugs.